src/stream.rs

```diff
@@ -25,29 +27,35 @@ unsafe impl Sync for Decompress {}
 /// This stream is at the heart of the I/O encoders and is used to compress
 /// data.
 pub struct Compress {
-    state: *mut brotli_sys::RustBrotliCompressor,
+    state: *mut brotli_sys::BrotliEncoderState,
 }
 
 unsafe impl Send for Compress {}
 unsafe impl Sync for Compress {}
 
 /// Parameters passed to various compression routines.
 pub struct CompressParams {
-    params: *mut brotli_sys::RustBrotliParams,
+    /// Compression mode.
+    pub mode: CompressMode,
+    /// Controls the compression-speed vs compression-density tradeoffs. The higher the `quality`,
+    /// the slower the compression. Range is 0 to 11.
+    pub quality: u32,
+    /// Base 2 logarithm of the sliding window size. Range is 10 to 24.
+    pub lgwin: u32,
+    /// Base 2 logarithm of the maximum input block size. Range is 16 to 24. If set to 0, the value
+    /// will be set based on the quality.
+    pub lgblock: u32,
```

Comment:
I personally prefer to leave these as private and allow mutation through the builder-style interface. That way we can add fields later on without breaking backwards compatibility